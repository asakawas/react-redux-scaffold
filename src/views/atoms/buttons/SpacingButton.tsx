import Button, { ButtonProps } from '@material-ui/core/Button';
import styled from 'styled-components';
import { spacing, SpacingProps } from '@material-ui/system';

/**
 * The type of props of SpacingButton.
 */
export type SpacingButtonProps = Readonly<SpacingProps & ButtonProps>;

const SpacingButton = styled(Button)`
  ${spacing}
`;

export default SpacingButton;
