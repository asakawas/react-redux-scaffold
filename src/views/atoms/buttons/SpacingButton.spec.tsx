import React from 'react';
import { render } from '@testing-library/react';
import SpacingButton from './SpacingButton';

describe('SpacingButton', () => {
  test('is rendered as expected', () => {
    const { asFragment } = render(
      <SpacingButton ml={1} pb={2}>
        Test
      </SpacingButton>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
