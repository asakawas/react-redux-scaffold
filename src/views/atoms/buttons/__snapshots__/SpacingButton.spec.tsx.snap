// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpacingButton is rendered as expected 1`] = `
<DocumentFragment>
  .c0 {
  margin-left: 8px;
  padding-bottom: 16px;
}

<button
    class="MuiButtonBase-root MuiButton-root MuiButton-text c0"
    ml="1"
    pb="2"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Test
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
